src/panels/config/js/script/device.js

```diff
@@ -0,0 +1,55 @@
+import { h, Component } from "preact";
+
+import "../../../../components/device/ha-device-picker";
+import "../../../../components/device/ha-device-action-picker";
+
+import { onChangeEvent } from "../../../../common/preact/event";
+
+export default class DeviceAction extends Component {
+  constructor() {
+    super();
+    this.onChange = onChangeEvent.bind(this, "action");
+    this.devicePicked = this.devicePicked.bind(this);
+    this.deviceActionPicked = this.deviceActionPicked.bind(this);
+    this.state.device_id = undefined;
+  }
+
+  devicePicked(ev) {
+    this.setState({ device_id: ev.target.value });
+  }
+
+  deviceActionPicked(ev) {
+    const deviceAction = ev.target.value;
+    this.props.onChange(this.props.index, (this.props.action = deviceAction));
```

Comment:
we should never mutate objects in Preact. Always make copies.

Reply:
OK, i think it's fixed now?